### PR TITLE
WT-13733 Split the reconcile header file into two seperate headers

### PR DESCRIPTION
**Pull request by kommiharibabu**
Now the reconcile module has two header files.
1. reconcile.h
2. reconcile_private.h.

The reconcile.h has all the public members and the reconcile_private.h has all the private members of the reconcile module.

### CODE REVIEW

**Comment by Siddhartha8899** [comment]
Except `WT_DELETE_HS_UPD`, `WT_RECONCILE` and `WT_OP_CONTEXT` I was able to move everything to reconcile.private.h, is there a reason why the other symbols were not moved?

**Comment by Siddhartha8899** [comment]
Perhaps we should keep the following at end of the `reconcile.h` file?

```
#ifdef HAVE_UNITTEST

#endif
```

**Review by Siddhartha8899 — APPROVED** [review]
LGTM!

File: dist/prototypes.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 # instead of in src/include/. As a result all relevant code is contained to a single folder. 
 # Adding modules to this list will automatically generate those headers, and we expect this list to 
 # grow as we modularise the code base.
-SELF_CONTAINED_MODULES = ["log", "evict"]
+SELF_CONTAINED_MODULES = ["log", "evict", "reconcile"]
```
Comment (Siddhartha8899) [inline dist/prototypes.py:26]: We can make this list alphabetical as part of this PR.

Comment (kommiharibabu) [reply dist/prototypes.py]: Yes, done. Thanks.

**Review by zunyiliu — APPROVED** [review]
LGTM!

**Pull request merged**